frontend/packages/core/src/Contexts/storage-context/reducer.tsx

```diff
@@ -0,0 +1,75 @@
+import { removeLocalData, rotateDataFromAPI, storeLocalData } from "./helpers";
+import type { Action, StorageState, UserPayload } from "./types";
+
+/**
+ * Reducer for the StorageContext
+ * This will act on the StorageState and add / remove items from the temporary storage
+ * as well as localStorage, this will keep all storage actions in one location for all
+ * components and allow for easier state hydration across the entire application
+ */
```

Comment:
I'm not sure a global storage context is the right direction for the frontend. We use data-layouts per workflow (instead of something like Redux) to keep data isolated and simplify the understanding of data as it moves through the application. Sharing storage between components has the potential to become untenable if used improperly. Data is also destroyed when a workflow is unmounted (unless it's kept in local storage of course), which I think this might change in some cases.

I can see the value in a storage helper that comes from a hook that returns a scoped instance of storage so we don't have to repeat the same complex code everywhere to manage storage.

I can also see the value in a global storage context. But I think we should discuss a fair bit more before introducing it to  the frontend, and it's not necessary for the shortlink feature as it is scoped today (as far as I'm aware).

Definitely curious to get more of your thoughts on this!

Reply:
Lets sync in person, will set up some time to go over this.